packages/@aws-cdk/assertions/lib/match.ts

```diff
@@ -304,8 +304,33 @@ class SerializedJson extends Matcher {
     super();
   };
 
+  private serializeCfnJoinGettAtt(object: any): any {
+    if (typeof object === 'object') {
+      if (object === null) return null;
+      if (object instanceof Array) {
+        for (var i = 0; i < object.length; i++) {
+          object[i] = this.serializeCfnJoinGettAtt(object[i]);
+        }
+      } else {
+        if ( object['Fn::GetAtt'] instanceof Array) {
+          return `<Fn::GetAtt:${object['Fn::GetAtt'][0]}:${object['Fn::GetAtt'][1]}>`;
+        } else if ( object['Fn::Join'] instanceof Array && object['Fn::Join'].length == 2 && object['Fn::Join'][1] instanceof Array ) {
+          const delimiter = object['Fn::Join'][0];
+          const valuesList = object['Fn::Join'][1];
+          return valuesList.map( (attr) => this.serializeCfnJoinGettAtt(attr)).join(delimiter);
+        } else {
+          for (var property in object) {
+            object[property] = this.serializeCfnJoinGettAtt(object[property]);
+          }
+        }
+      }
+    }
+    return object;
+  }
+
   public test(actual: any): MatchResult {
     const result = new MatchResult(actual);
+    actual = this.serializeCfnJoinGettAtt(actual);
```

Comment:
I think we should separate this into two functions, `resolveCfnJoin` and `resolveCfnGetAtt`

Reply:
This would double the cost of the function, as you have to recursively visit every node twice. Could imagine to extend this functionality to handle all intrinsic functions. But at the moment, i only found Joins ad GetAtts in my use cases.